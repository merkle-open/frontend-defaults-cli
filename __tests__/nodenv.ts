import { defaultIt } from './shared';

describe('nodenv', async () => {
	it('default', async () => await defaultIt('nodenv', '--nodenv --force --noInstall'));
	it('with-githooks', async () => await defaultIt('nodenv-with-githooks', '--nodenv --githooks --force --noInstall'));
});
