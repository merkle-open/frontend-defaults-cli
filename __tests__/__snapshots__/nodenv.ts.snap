// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nodenv default 1`] = `
"{
  \\"ts\\": false,
  \\"tslint\\": false,
  \\"es\\": false,
  \\"eslint\\": false,
  \\"editorconfig\\": false,
  \\"prettier\\": false,
  \\"stylelint\\": false,
  \\"gitignore\\": false,
  \\"npmrc\\": false,
  \\"readme\\": false,
  \\"githooks\\": false,
  \\"commitlint\\": false,
  \\"nodenv\\": true,
  \\"webpack\\": false,
  \\"install\\": false,
  \\"force\\": true
}"
`;

exports[`nodenv default 2`] = `
"8.12.0
"
`;

exports[`nodenv default 3`] = `
"{
  \\"devDependencies\\": {
    \\"npm-run-all\\": \\"4.1.3\\"
  },
  \\"scripts\\": {
    \\"lint\\": \\"npm-run-all lint:*\\"
  }
}"
`;

exports[`nodenv with-githooks 1`] = `
"{
  \\"ts\\": false,
  \\"tslint\\": false,
  \\"es\\": false,
  \\"eslint\\": false,
  \\"editorconfig\\": false,
  \\"prettier\\": false,
  \\"stylelint\\": false,
  \\"gitignore\\": false,
  \\"npmrc\\": false,
  \\"readme\\": false,
  \\"githooks\\": true,
  \\"commitlint\\": false,
  \\"nodenv\\": true,
  \\"webpack\\": false,
  \\"install\\": false,
  \\"force\\": true
}"
`;

exports[`nodenv with-githooks 2`] = `
"# ~/.huskyrc
# Note: it will run in /bin/sh shell
if command -v nodenv >/dev/null 2>&1; then
  eval \\"$(nodenv init -)\\"
fi"
`;

exports[`nodenv with-githooks 3`] = `
"8.12.0
"
`;

exports[`nodenv with-githooks 4`] = `
"{
  \\"devDependencies\\": {
    \\"npm-run-all\\": \\"4.1.3\\"
  },
  \\"scripts\\": {
    \\"lint\\": \\"npm-run-all lint:*\\"
  }
}"
`;
